backend/daemon/daemon_e2e_test.go

```diff
@@ -30,56 +32,76 @@ func TestAPIGetRemotePublication(t *testing.T) {
 	ctx := context.Background()
 
 	// Carol will be the DHT server
-	carol := makeTestApp(t, "carol", makeTestConfig(t), true)
+	dhtProvider := makeTestApp(t, "carol", makeTestConfig(t), true)
 
-	var alice *App
-	{
-		cfg := makeTestConfig(t)
-		cfg.P2P.BootstrapPeers = carol.Net.MustGet().Libp2p().AddrsFull()
-		alice = makeTestApp(t, "alice", cfg, true)
-	}
+	requester, publishedDocument, _ := makeRemotePublication(t, ctx, dhtProvider)
 
-	var bob *App
-	{
-		cfg := makeTestConfig(t)
-		cfg.P2P.BootstrapPeers = carol.Net.MustGet().Libp2p().AddrsFull()
-		bob = makeTestApp(t, "bob", cfg, true)
-	}
+	remotePublication, err := requester.RPC.Documents.GetPublication(ctx, &documents.GetPublicationRequest{DocumentId: publishedDocument.Document.Id})
+	require.NoError(t, err)
+	testutil.ProtoEqual(t, publishedDocument, remotePublication, "remote publication doesn't match")
+}
 
-	// Make sure bob does't know anything about alice.
-	require.NoError(t, bob.Net.MustGet().Libp2p().Network().ClosePeer(alice.Repo.Device().ID()))
-	bob.Net.MustGet().Libp2p().Peerstore().RemovePeer(alice.Repo.Device().ID())
+func TestGateway(t *testing.T) {
```

Comment:
Consider moving this test to the gateway package. Main packages can also have test files. This seems to be very specific to the gateway.

Reply:
It was there to reuse unexported `makeTestApp` `makeTestConfig` and `makeRemotePublication`. Also because the gateway is kinda a daemon mode of operation.